### DISCUSSION TITLE
Open3D ICP on Cuda Device is slow and doesn't seem to run on the GPU at all

### DISCUSSION THREAD

**Discussion by JonasHuu** [isl-org/Open3D#6431] [Q&A]
Hello everyone,

I've just built Open3D from source including the Cuda module on a Windows 10 PC which seemed to work fine without any errors. I'm trying to perform multi scale colored ICP registration for multiple point clouds and wanted to make this process quicker by using my GPU instead of my CPU. Now as I understand, the only thing I need to do for the function o3d.t.pipelines.registration.multi_scale_icp to use the GPU is to create the source and target point cloud on the GPU with e.g. source.cuda(0) or specifying the device when creating it (I'm reading it from a .ply file so I'm doing it with source.cuda(0)). As estimation, I'm using the TransformationEstimationForColoredICP() method. This works, but it is slower than simply leaving the point clouds on the CPU and when I open my task manager during that process, the GPU is not being used at all while the CPU is working hard. Do you guys know if that's a problem with trying to use Windows or a general problem with the library? Or am I doing something wrong?

The point cloud looks like this:
PointCloud on CUDA:0 [263723 points (Float32)].
Attributes: normals (dtype = Float32, shape = {263723, 3}), colors (dtype = Float32, shape = {263723, 3})

**Comment by JonasHuu** [comment]
The reason why the GPU version was slower than the CPU one is my GPU (Quadro T1000). When e.g. using a RTX A4000, it takes a lot less time (0.5s instead of 3s). It was useful using the Nsight tool to actually see that it runs on the GPU since the task manager just is not a suitable tool for that or simply the TechPowerUp GPU-Z tool.

**Comment by FraDePace** [comment]
Which version of Open3D did you build? v0.18.0? 
Can you please share how you built it with CUDA on Windows? also which CUDA and CUDNN versions you used?

I am struggling with v0.18.0
